axt/axt.go

```diff
@@ -44,30 +44,40 @@ func Read(filename string) []Axt {
 	return answer
 }
 
-// ReadToChan takes a filename and a channel.  The function will read all remaining Axt records
+// ReadToChan takes a filename, a channel for the axt structs, and a channel for
+// the header text.  The function will read all remaining Axt records
 // from the open file into the channel.  The function will then close the file and close the channel.
-func ReadToChan(filename string, data chan<- Axt) {
+func ReadToChan(filename string, data chan<- Axt, header chan<- []string) {
 	var file *fileio.EasyReader
 	var curr Axt
 	var done bool
 	var err error
+	var headerLines []string
 
 	file = fileio.EasyOpen(filename)
+
+	headerLines, err = fileio.EasyReadHeader(file)
+	exception.PanicOnErr(err)
+	header <- headerLines
+	close(header)
+
 	for curr, done = ReadNext(file); !done; curr, done = ReadNext(file) {
 		data <- curr
 	}
+
 	err = file.Close()
 	exception.PanicOnErr(err)
 	close(data)
 }
 
-// GoReadToChan takes a filename and returns a channel.  GoReadToChan
+// GoReadToChan takes a filename and returns a channel and header text.  GoReadToChan
 // will launch a Go routine to read all the alignments from the file into the channel
 // and then close the channel.
-func GoReadToChan(filename string) <-chan Axt {
+func GoReadToChan(filename string) (<-chan Axt, []string) {
```

Comment:
Yes, you are correct that we could pass back the channel (from which a programmer could read the header) instead of the header itself.  Since we know that there will only be one header, we go ahead and read the header and pass back this one header, which saves the programmer calling this function some time (since we already "know" what they will do).  Would work equally well to pass back a channel to a header and we would likely do that if there were more than one header per file (especially if it is an unknown number), similar to passing back a channel for the Axts, where we don't know how many there will be and there will likely be many of them.